jsonschema2pojo-integration-tests/src/test/java/org/jsonschema2pojo/integration/util/JsonSchema2PojoRule.java

```diff
@@ -0,0 +1,237 @@
+import org.junit.runners.model.Statement;
+import static org.apache.commons.io.FileUtils.*;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * A JUnit rule that executes JsonSchema2Pojo.
+ *
+ * @author Christian Trimble
+ *
+ */
+public class JsonSchema2PojoRule implements TestRule {
+
+    private File sourcesDir;
+    private File classesDir;
+    private boolean inRule = false;
+    private boolean cleanSourcesDir = true;
+    private boolean cleanClassesDir = true;
+    private ClassLoader classLoader;
+
+    public File getSourcesDir() {
+        initializeSourcesDir();
+        return sourcesDir;
+    }
+
+    public File getClassesDir() {
+        initializeClassesDir();
+        return classesDir;
+    }
+
+    public ClassLoader getClassLoader() {
+        return classLoader;
+    }
+
+    @Override
+    public Statement apply(final Statement base, final Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                inRule = true;
+                try {
+                    sourcesDir = methodNameDir(classNameDir(sourceBaseDir(), description.getClassName()),
+                            description.getMethodName());
+                    classesDir = methodNameDir(classNameDir(classBaseDir(), description.getClassName()),
+                            description.getMethodName());
+
+                    base.evaluate();
+                } finally {
+                    sourcesDir = null;
+                    classesDir = null;
+                    classLoader = null;
+                    cleanSourcesDir = true;
+                    cleanClassesDir = true;
+                    inRule = false;
+                }
+            }
+        };
+    }
+
+    public File generate(String schema, String targetPackage) {
+        Map<String, Object> configValues = Collections.emptyMap();
+        return generate(schema, targetPackage, configValues);
+    }
+
+    public File generate(URL schema, String targetPackage) {
+        Map<String, Object> configValues = Collections.emptyMap();
+        return generate(schema, targetPackage, configValues);
+    }
+
+    public File generate(String schema, String targetPackage, Map<String, Object> configValues) {
+        URL schemaUrl = CodeGenerationHelper.class.getResource(schema);
+        assertThat("Unable to read schema resource from the classpath: " + schema, schemaUrl, is(notNullValue()));
+
+        return generate(schemaUrl, targetPackage, configValues);
+    }
+
+    public File generate(final URL schema, final String targetPackage, final Map<String, Object> configValues) {
+        if (!inRule) {
+            throw new RuntimeException("cannot generate outside of rule");
+        }
+        initializeSourcesDir();
+        CodeGenerationHelper.generate(schema, targetPackage, configValues, sourcesDir);
```

Comment:
Instead of calling `initializeSourceDir()` here, just use the accessor to do the needed checks.